src/main/java/net/sf/mzmine/modules/peaklistmethods/gapfilling/samerange/SameRangeTask.java

```diff
@@ -55,179 +59,246 @@
     private boolean removeOriginal;
 
     private int processedRows, totalRows;
+    private AtomicInteger processedRowsAtomic;;
 
     private ParameterSet parameters;
 
     SameRangeTask(MZmineProject project, PeakList peakList,
-	    ParameterSet parameters) {
+        ParameterSet parameters) {
 
-	this.project = project;
-	this.peakList = peakList;
-	this.parameters = parameters;
+    this.project = project;
+    this.peakList = peakList;
+    this.parameters = parameters;
 
-	suffix = parameters.getParameter(SameRangeGapFillerParameters.suffix)
-		.getValue();
-	mzTolerance = parameters.getParameter(
-		SameRangeGapFillerParameters.mzTolerance).getValue();
-	removeOriginal = parameters.getParameter(
-		SameRangeGapFillerParameters.autoRemove).getValue();
+    suffix = parameters.getParameter(SameRangeGapFillerParameters.suffix)
+        .getValue();
+    mzTolerance = parameters.getParameter(
+        SameRangeGapFillerParameters.mzTolerance).getValue();
+    removeOriginal = parameters.getParameter(
+        SameRangeGapFillerParameters.autoRemove).getValue();
 
     }
 
     public void run() {
 
-	logger.info("Started gap-filling " + peakList);
+        logger.info("Started gap-filling " + peakList);
 
-	setStatus(TaskStatus.PROCESSING);
+        setStatus(TaskStatus.PROCESSING);
 
-	// Get total number of rows
-	totalRows = peakList.getNumberOfRows();
+        // Get total number of rows
+        totalRows = peakList.getNumberOfRows();
 
-	// Get peak list columns
-	RawDataFile columns[] = peakList.getRawDataFiles();
+        // Get peak list columns
+        RawDataFile columns[] = peakList.getRawDataFiles();
 
-	// Create new peak list
-	processedPeakList = new SimplePeakList(peakList + " " + suffix, columns);
+        // Create new peak list
+        processedPeakList = new SimplePeakList(peakList + " " + suffix, columns);
 
-	// Fill gaps in given column
-	for (int row = 0; row < totalRows; row++) {
 
-	    // Canceled?
-	    if (isCanceled())
-		return;
+        /*************************************************************
+            Creating a stream to process the data in parallel
+        */
 
-	    PeakListRow sourceRow = peakList.getRow(row);
-	    PeakListRow newRow = new SimplePeakListRow(sourceRow.getID());
+        List<Integer> rowList = new ArrayList<Integer>();

```

Comment:
You could also directly create the parallel stream

`IntStream.range(0, totalRows).parallel().forEach`